packages/core/src/state/NetworkManager.ts

```diff
@@ -58,7 +59,14 @@ export default class NetworkManager implements Manager {
             // only receive after new state is computed
             return next(action).then(() => {
               if (action.meta.key in this.fetched) {
-                this.handleReceive(action);
+                // Note: meta *must* be set by reducer so this should be safe
+                const error = getState().meta[action.meta.key].error;
```

Comment:
if reducer turns this into an error, we should reject the promise. In the future perhaps we should *only* use this method to detect rejections, rather than having a separate code path for rejecting when the promise also rejects. this would greatly simplify the concept.